uwblueprint/supportive-housing
# Resident Directory Table and Pagination

## GitHub Issue link

[Resident Directory](https://github.com/uwblueprint/supportive-housing/issues/90)



## Implementation description
![Screen Shot 2023-07-12 at 9 09 59 AM](https://github.com/uwblueprint/supportive-housing/assets/53386475/e6dd7954-b882-4f61-bf33-129bf110f4ea)


- Implemented table component for Resident directory page
- Moved getResidents to `ResidentAPIClient`
- Updated params of getResidents service, route, and API client function
- Hooked up pagination component to resident directory table



## Steps to test
1. Create test residents in your local DB
2. Verify that test residents appear in table + pagination works as intended
3. Ensure that residents with no end date in the DB have the "Current" status in the table
4. Ensure that residents WITH an end date in the DB have the "Past" status in the table



## What should reviewers focus on?
- Column widths feel a bit wonky ; I adjusted the widths adhoc, but as per discussion with @Safewaan this is something we'll revisit later 
- Didn't implement the "Add Resident" button, but rather marked it as a TODO in with a comment ; we can add this back in if needed


## Checklist
